source/blood/src/sectorfx.cpp

```diff
@@ -298,9 +298,9 @@ void DoSectorPanning(void)
                 if (pSector->ceilingstat&64)
                     angle -= 512;
                 int xBits = (picsiz[nTile]&15)-((pSector->ceilingstat&8)!=0);
-                px += mulscale30(speed<<2, Cos(angle))>>xBits;
+                px += mulscale30(speed<<2, Cos(-angle))>>xBits;
```

Comment:
This makes more sense to me now. Is this logically the same as what you had, or was it bugged?

Reply:
Yeah, I had down wrong before. It fixed the test case in the issue but with other levels such as E2M1 the water was inverted. Turns out the calculations were fine, just ceiling inverts the angle.